rts/motoko-rts/src/idl.rs

```diff
@@ -539,8 +539,16 @@ unsafe fn is_opt_reserved(typtbl: *mut *mut u8, end: *mut u8, t: i32) -> bool {
     return t == IDL_CON_opt;
 }
 
+#[derive(Clone, Copy)]
+pub enum CompatibilityMode {
+    // Conventional Candid deserialization compatibility.
+    Candid,
+    // Memory compatibility for orthogonal persistence.
+    Persistence,
+}
+
 // TODO: consider storing fixed args typtbl1...end2 in `rel` to use less stack.
-unsafe fn sub(
+pub(crate) unsafe fn sub(
```

Comment:
Another possible pitfall here is that, in Candid blob = vec Nat8. For Motoko subtyping, we will need distinct primitive descriptors for Blob and Vec Nat8, i.e. a slightly different mapping from types to descriptors.

Reply:
Thanks a lot for pointing this additional differences out. I will address them. Very important points.

Reply:
**Update**: I addressed these cases with the new `memory_compatible()` function that is specifically used by the orthogonal persistence.

Reply:
I think duplicating and adapting the code works nicely. Thank you!